### PR TITLE
[cd] Update & push submodule in cdr/m on pushes to master

### PR DESCRIPTION
**Pull request by bpmct**
This should trigger a build of coder.com/docs automatically instead of waiting for dependabot.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -20,14 +19,13 @@ jobs:
       - name: Checkout Docs
         run: git fetch && git checkout ${GITHUB_SHA}
```
Comment (bpmct) [inline .github/workflows/deploy.yaml:20]: I wasn't sure if we wanted to do `git checkout ${GITHUB_SHA}` or `git checkout master`

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -20,14 +19,13 @@ jobs:
...
-          GITHUB_REF: ${{ github.ref }}
-          GITHUB_REPOSITORY: ${{ github.repository }}
-          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
-          VERCEL_ORG_ID: team_tGkWfhEGGelkkqUUm9nXq17r
-          VERCEL_PROJECT_ID: QmZRucMRh3GFk1817ZgXjRVuw5fhTspHPHKct3JNQDEPGd
+      - name: Commit files
+          run: |
+            git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git config --local user.name "github-actions[bot]"
+            git commit -m "cdr/docs: bump product/coder.com/site/docs ${GITHUB_SHA}" -a
```
Comment (bpmct) [inline .github/workflows/deploy.yaml:26]: We can definitely change these. I was following https://github.com/marketplace/actions/github-push

**Comment by kylecarbs** [comment]
I'm relatively hesitant to make this workflow automated... the fact that pushing content here could cause our monorepo to fail a build scares me.

What's the reason behind this change? If we need docs to update faster we could manually trigger dependabot in an action so changes get merged faster.

**Comment by cmoog** [comment]
> I'm relatively hesitant to make this workflow automated... the fact that pushing content here could cause our monorepo to fail a build scares me.

So we could always revert the commit in the mono repo.... the integrity of version control in the mono repo is maintained with this change.

**Comment by cmoog** [comment]
> What's the reason behind this change? If we need docs to update faster we could manually trigger dependabot in an action so changes get merged faster.

It would be cleaner for contributors if they didn't have to think about the mono repo at all.

**Comment by kylecarbs** [comment]
Agreed agreed

**Comment by bpmct** [comment]
Yeah, I don't like the idea that something like a typo fix could take up to 24h (+ approve time) to update on the site. Ideally, the docs should be automatically built after a push. 

We could potentially do this without pushing code to master on the monorepo, but then we probably shouldn't be importing our docs into the monorepo as a submodule anyways.

**Review by cmoog — APPROVED** [review]
The approach looks great to me 👍 . I assume this is something we can only _really_ test by merging.

**Review by kylecarbs — APPROVED** [review]

**Pull request merged**